build/changelog/ChangelogBuilder.php

```diff
@@ -0,0 +1,165 @@
+<?php
+namespace Aws\Build\Changelog;
+
+/**
+ * @internal
+ */
+class ChangelogBuilder
+{
+
+    private $newServiceFlag = false;
+    private $baseDir;
+    private $releaseNotesOutputDir;
+    private $verboseFlag;
+
+    public function __construct($baseDir, $releaseNotesOutputDir, $verboseFlag)
+    {
+        $this->baseDir = $baseDir;
+        $this->releaseNotesOutputDir = $releaseNotesOutputDir;
+        $this->verboseFlag = $verboseFlag;
+    }
+
+    private function readChangelog()
+    {
+        $releaseDir = $this->baseDir . '.changes/nextrelease/';
+        $changelogEntries = [];
+        if (!is_dir($releaseDir) || !$dh = opendir($releaseDir)) {
+            throw new \InvalidArgumentException("nextrelease directory doesn't exists or is not readable at location $releaseDir");
+        } else {
+            //Ignore any files starting with a (.) dot
+            $files = preg_grep('/^([^.])/', scandir($releaseDir));
+            if (empty($files)) {
+                throw new \InvalidArgumentException("No release notes files found in $releaseDir folder");
+            }
+            foreach ($files as $file) {
+                $str = file_get_contents($releaseDir . $file);
+                $changelogEntries = array_merge($changelogEntries, $this->cleanJSON(json_decode($str)));
+            }
+            closedir($dh);
+        }
+        return $changelogEntries;
+    }
+
+    private function cleanJSON($arr)
+    {
+        if (empty($arr) || !is_array($arr)) {
+            throw new \Exception('Invalid Input', 2);
+        } else {
```

Comment:
same, might not necessarily needs to be inside the else block, if exception occurs, should have been stopped anyway